### PR TITLE
refactor: updated NumberInput Component

### LINKED ISSUE #3366: Separation of UI and Logic in the `NumberInput` Component
### Description

Separate logic into separate files such as `use-xxx.ts`.
In addition, modify the hooks to function on their own.

### Additional Information

Create a new branch from `v2.0` branch for development.
https://react-spectrum.adobe.com/react-aria/useNumberField.html#usenumberfield

### PR DESCRIPTION
**Pull request by ogison**
Closes #3366

## Description

Separation of UI and Logic in the NumberInput Component 

## New behavior
No

### CODE REVIEW

File: .changeset/happy-pans-juggle.md
Comment (108yen) [inline .changeset/happy-pans-juggle.md:1]: Please remove this file.
Changeset is not needed in this change. Because this change is just refactoring.

**Review by 108yen — CHANGES_REQUESTED** [review]
Target branch is wrong.
Change to `v2.0`.

**Comment by ogison** [comment]
I’ve updated the target branch to v2.0.

File: .changeset/happy-pans-juggle.md
Comment (ogison) [reply .changeset/happy-pans-juggle.md:1]: This has been addressed.
Please review it at your convenience.

**Review by 108yen — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request closed without merging**